### DISCUSSION TITLE
Warnings/Errors parsing smart playlists

### DISCUSSION THREAD

**Discussion by john-** [jcorporation/myMPD#1346] [Q&A] [answered]
At some point in the past I enabled smart playlists but never really used them.  Recently I noticed warnings/errors in the mympd log in the area of parsing the smart playlists:
```
[snip many log entries including smart playlist parser warnings]
Sep 18 08:22:09 frontaudio mympd[540]: WARN     mpdworker       $.sortdesc: JSON path "$.sortdesc" not found
Sep 18 08:22:09 frontaudio mympd[540]: WARN     mpdworker       $.maxentries: JSON path "$.maxentries" not found
Sep 18 08:22:10 frontaudio mympd[540]: WARN     mpdworker       $.sort: JSON path "$.sort" not found or value is not string type, found type is "MJSON_TOK_INVALID"
Sep 18 08:22:10 frontaudio mympd[540]: WARN     mpdworker       $.value: JSON path "$.value" not found or value is not string type, found type is "MJSON_TOK_INVALID"
Sep 18 08:22:10 frontaudio mympd[540]: ERROR    mpdworker       Can't parse smart playlist file "/var/lib/mympd/smartpls/myMPDsmart-mostPlayed" (sticker)
Sep 18 08:22:10 frontaudio mympd[540]: NOTICE   mpdworker       11 smart playlists updated, 281 already up-to-date
Sep 18 08:22:10 frontaudio mympd[540]: NOTICE   mpdworker       Stopping mpd_worker thread
```
Also, those with errors were not populating in the UI

I fixed one of the errors (myMPDsmart-bestRated) by manually changing it to match the example in the [documentation](https://jcorporation.github.io/myMPD/references/smart-playlists).  After those changes it populated in the UI and I could edit it and view the songs.

However, it doesn't make sense that I have to manually fix all the rest as these json files seem like files generated/installed by mympd.  Maybe I have old versions that do not contain correct values?  Is there a straightforward way to get working versions?

**Comment by jcorporation** [comment, marked as answer]
In the maintenance there is an option to delete all smart playlists. Smart playlists generated through the smart playlists generation feature are the correctly recreated on next maintenance run.

**Reply by john-** [reply]
Thanks.  That addressed all the warnings.

However, I see that 2 of the smart playlists I had were not recreated through the regeneration feature: `myMPDsmart-bestRated` and `myMPDsmart-mostPlayed`.  Were these ones I manually created previously so were not regenerated?

**Reply by jcorporation** [reply]
Correct you can add it manually.

**Reply by john-** [reply]
Thanks!
